coredns/endpointslice/map.go

```diff
@@ -111,6 +122,41 @@ func (m *Map) Put(es *discovery.EndpointSlice) {
 		return
 	}
 
+	// TODO: Only do this look up if globalnet enabled to reduce unnessary call to Kube API
+	var localSlice *discovery.EndpointSlice
+	if cluster == m.localClusterID {
+		clientSet, err := kubernetes.NewForConfig(m.kubeConfig)
+		if err != nil {
+			klog.Warningf("Error creating client set %#v", err)
+			return
+		}
+
```

Comment:
It would be nice to add unit tests to cover this new block of code but we can do that in another PR.